src/silhan/Competition.java

```diff
@@ -0,0 +1,153 @@
+package silhan;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.NoSuchElementException;
+
+public class Competition {    
+    //data
+    private String name;
+    private ArrayList<Competitor> competitors;
+    private int counter=1;
+    
+    //konstruktor
+    public Competition(String name){
+        this.name = name;
+        this.competitors = new ArrayList<>(); //konstruktor
+    }
+    
+    //gettery
+    public String getName() {
+        return name;
+    }
+
+    public int getCounter() {
+        return counter;
+    }
+    
+    //defensive deep copy
+    public ArrayList<Competitor> getCompetitors() {
+        ArrayList<Competitor> copy = new ArrayList<>();
+        for (Competitor competitor : competitors){ //prochazim pole competitors (v kazdem kroku jednoho zavodnika)
+            copy.add(new Competitor(competitor)); //volam kopirujici konstruktor a pridavam do pole copy
+        }
+        return copy;
+    }
+
+    public int getnCompetitors() {
+        return competitors.size();
+    }
+    
+    //methods    
+    public void addCompetitor(String name, String surname, int year, char gender){
+        competitors.add(Competitor.getInstance(name, surname, year, gender, this.getCounter()));
+        counter++;
+    }
+    
+    public void setStartTimeAll(int hours, int min, int sec){
+        for (Competitor competitor : competitors) {
+            competitor.setStartTime(hours, min, sec);
+        }
+    }
+    
+    //zavodnici startuji s casovym rozestupem podle registracnich cisel
```

Comment:
tu je to skôr podľa indexu v "štartovnom" poli - registračné číslo by mohli mať aj iné